src/js/components/Calendar/__tests__/Calendar-test.js

```diff
@@ -149,4 +164,297 @@ describe('Calendar', () => {
     expect(onSelect).toBeCalledWith(expect.stringMatching(/^2018-01-17T/));
     expect(container.firstChild).toMatchSnapshot();
   });
+
+  test('first day sunday week monday', () => {
+    // the first day of the month is on Sunday but the
+    // user wants the week to start on Monday
+    const { container } = render(
+      <Grommet>
+        <Calendar
+          firstDayOfWeek={1}
+          date="2020-03-01T00:00:00-08:00"
+          animate={false}
+        />
+      </Grommet>,
+    );
+
+    expect(container.firstChild).toMatchSnapshot();
+  });
+
+  test('animate', () => {
+    jest.useFakeTimers();
+    const { container, getByLabelText } = render(
+      <Grommet>
+        <Calendar date={DATE} />
+      </Grommet>,
+    );
+    fireEvent.click(getByLabelText('December 2017'));
+    act(() => {
+      jest.advanceTimersByTime(400);
+    });
+    fireEvent.click(getByLabelText('January 2018'));
+    act(() => {
+      jest.advanceTimersByTime(400);
+    });
+    expect(container.firstChild).toMatchSnapshot();
+  });
+
+  test('onEnter', async () => {
+    const onSelect = jest.fn();
+    const { container, getByText } = render(
+      <Grommet>
+        <Calendar
+          bounds={['2018-01-14', '2018-01-16']}
+          date={DATE}
+          onSelect={onSelect}
+          animate={false}
+        />
+      </Grommet>,
```

Comment:
This looks like a potentially reusable code that appears in multiple tests.
Also, can we update the dates to 2020?

Reply:
I updated the dates to 2020 and setup the <Calendar>...</Calendar> code to be reusable across the tests that have it